src/box/lua/iproto.c

```diff
@@ -287,23 +166,32 @@ push_iproto_protocol_features(struct lua_State *L)
 	lua_pushinteger(L, IPROTO_CURRENT_VERSION);
 	lua_setfield(L, -2, "protocol_version");
 
-	const struct iproto_constant features[] = {
-		{"streams", IPROTO_FEATURE_STREAMS},
-        	{"transactions", IPROTO_FEATURE_TRANSACTIONS},
-        	{"error_extension", IPROTO_FEATURE_ERROR_EXTENSION},
-        	{"watchers", IPROTO_FEATURE_WATCHERS},
-        	{"pagination", IPROTO_FEATURE_PAGINATION},
-	};
+	size_t region_svp = region_used(&fiber()->gc);
+	size_t alloc_size;
+	struct iproto_constant *features = xregion_alloc_array(
+		&fiber()->gc, struct iproto_constant,
+		iproto_feature_id_constants_size, &alloc_size
+	);
+	static const char prefix[] = "IPROTO_FEATURE_";
+	size_t prefix_len = sizeof(prefix) - 1;
+	for (size_t i = 0; i < iproto_feature_id_constants_size; ++i) {
+		const char *name = iproto_feature_id_constants[i].name;
+		assert(strncmp(name, prefix, prefix_len) == 0);
+		features[i].name = strtolowerdup(name + prefix_len);
```

Comment:
Looks like you leak memory here.
![image](https://user-images.githubusercontent.com/32464072/235200596-7220f3ec-c81b-49f1-bfd5-479532e2c342.png)

Reply:
Thanks, fixed